controllers/helpers.go

```diff
@@ -146,3 +146,35 @@ func ErrorStatusUpdater(reason string) StatusUpdater {
 
 func DoNothingStatusUpdater(ins aws.Instance, obj AWSObjectStatusResource, ctx context.Context, sw client.StatusWriter, log logr.Logger) {
 }
+
+func CleanUpPolicyVersions(svc iamiface.IAMAPI, policyARN string) error {
+	maxVersions := 4
+
+	if len(policyARN) < 20 {
+		return nil
+	}
+
+	resp, err := svc.ListPolicyVersions(&awsiam.ListPolicyVersionsInput{
+		PolicyArn: &policyARN,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	if len(resp.Versions) <= maxVersions {
+		return nil
+	}
+
+	// We need to delete oldest versions
+	for i := len(resp.Versions) - 1; i >= maxVersions; i-- {
```

Comment:
What's the point of the for loop? Shouldn't be enought by referencing the oldest element of the array directly? (as we only want to remove one)

Reply:
We need to have max 4 versions, the correct behavior is remove only one but to be sure that we have max 4 I added this loop. It should not happen but the policy can be updated from AWS console and it creates another version

Reply:
The max can only be 5, and the length of the array will never be more than 5. If the length of array is 5, run delete on the element with the lowest version number? I still don't understand the need of loop

Reply:
Ok, I see, I was thinking that we can have more than 5 versions but, obviously, we can have max 5 :sweat_smile:... So, yes, I'll change it to remove the fifth version in case it exist

Reply:
Here the change: https://github.com/redradrat/aws-iam-operator/pull/34/commits/c51925328d7114388f906cb4898894d1816f85be